### ISSUE TITLE
Newbie question

### ISSUE THREAD

**Issue by supra411** [bitkey/bitkey#31]
Hi:

Sorry, I tried googling it, but I can't seem to get bitkey to run on my macbook. It works on my windows pc. I'm booting it with fat32 file format.

Thanks!

**Comment by JedMeister** [comment]
Hi,

TBH, I'm not totally sure, but my guess is that it's to do with the UEFI BIOS on the Mac. From what I've heard, our installer doesn't properly support UEFI. We hope to fix that within the next few months.

**Comment by bogomip2k** [comment]
I've run into this same issue.

**Comment by ajgarnerin** [comment]
I have the same problem. Any progress on this issue?

**Comment by JedMeister** [comment]
Not really... but [soon](https://github.com/turnkeylinux/tracker/issues/942) hopefully?!

**Comment by sepehr** [comment]
Eagerly looking forward to this.

**Comment by bogomip2k** [comment]
This one is really big. Any ideas as to how much effort this would take to implement?

On Nov 18, 2017, at 3:38 PM, Sepehr Lajevardi <notifications@github.com<mailto:notifications@github.com>> wrote:


Eagerly looking forward to this.

**Comment by sepehr** [comment]
TBH I'm not very familiar with the topic to provide an estimate. However, I don't think this would be too much of [work](https://glandium.org/blog/?p=2830) for someone who is. As it seems to be [already solved](https://tails.boum.org/blueprint/UEFI_boot_on_Mac_without_rEFInd/).

**Comment by JedMeister** [comment]
Progress on this issue has ground to a halt lately. That's because the guy we have working on it is extremely busy and hasn't had much of a chance to work on it. I'm hoping that we will ditch our custom installer and move to the default Debian installer. If we do that, then hopefully this issue will essentially resolve itself.

But even in a best case scenario, that will be a little while away. And as BitKey is a side project, it will probably be one of the last to be updated. Although once we have applied it in the main repo, then it shouldn't be too hard for any of you to rebuild BitKey with this included yourselves.

So TL;DR - I doubt this will be working this year. It may be possible to build it yourself early next year, but it may be as late as Q2 (or even later) before we publicly release a BitKey compatible with UEFI.

Sorry for the bad news...

**Comment by sepehr** [comment]
That's totally understandable. I'm patiently looking forward to it.

**Comment by FromLV** [comment]
Does anyone checked the problem is solved?

**Comment by pltlg** [comment]
Seems not yet

**Comment by JedMeister** [comment]
No and TBH, it's unlikely that it will be solved anytime soon (at least by @turnkeylinux). 

If someone wants to have a crack at it, from my brief reading it should be possible to use grub (v2) to boot the ISO when a UEFI BIOS is found (currently it uses syslinux which is incompatible). So ideally we'd have a hybrid boot system where grub is used if required, otherwise it falls back to syslinux.

Implementation is the tricky bit though...

**Comment by estevaocm** [comment]
I have no experience with a Mac, but I have a UEFI PC notebook in which I was able to disable UEFI for legacy booting. No idea if this is something that can be accomplished on a Mac without modifying Bitkey.

**Comment by jologi** [comment]
why is bitkey using turnkeylinux and not ubuntu? Ubuntu will boot from Mac

**Comment by JedMeister** [comment]
@bogomip2k - because BitKey is a TurnKey Linux side project... :smile:

**Comment by estevaocm** [comment]
Also Ubuntu is not nearly as clean and secure as Debian. But by all means, @jologi, be our guest and build a Ubuntu version if you prefer. But I don't know if there is a building tool for Ubuntu that is as simple to use as Turnkey.

**Comment by QuantBox723** [comment]
I've the same problem with my Surface Pro, which has UEFI Bios. The temporary solution inspired by the following discussion [https://askubuntu.com/questions/395879/how-to-create-uefi-only-bootable-usb-live-media?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa](url), which worked for me, is:
(1) create a bootable Ubuntu installation USB stick and try if it boots (I used Ubuntu 18.04)
(2) delete all directories on USB stick except 
     .disk
     boot
     EFI
(BTW, if you want also to be able to boot using legacy BIOS leave in addttion isolinux and uui)
(3) extract casper directory from BitKey ISO, e.g. using 7zip, and copy it to your USB stick.
(4) replace boot\grub\grub.cfg configuration file with the following one
`
if loadfont /boot/grub/font.pf2 ; then
	set gfxmode=auto
	insmod efi_gop
	insmod efi_uga
	insmod gfxterm
	terminal_output gfxterm
fi

set menu_color_normal=white/black
set menu_color_highlight=black/light-gray

set timeout=10
menuentry "BitKey cold-offline (green): create wallet, sign transactions" {
	set gfxpayload=keep
	linux /casper/vmlinuz boot=casper root=/dev/ram rw showmounts toram bitkey=cold-offline  -- 
	initrd /casper/initrd.gz 
}
menuentry "BitKey cold-online (blue): watch wallet, prepare transactions" {
	set gfxpayload=keep
	linux /casper/vmlinuz boot=casper root=/dev/ram rw showmounts toram bitkey=cold-online --
	initrd /casper/initrd.gz
}
menuentry "BitKey hot-online (red): create & watch wallet, prepare & sign transactions" {
	set gfxpayload=keep
	linux /casper/vmlinuz boot=casper root=/dev/ram rw showmounts toram bitkey=hot-online --
	initrd /casper/initrd.gz
}
`

Thats it.
This workaround is not working perfectly  however: BitKey does not recognize Type Cover keyboard and touchpad on my Surface, so that external usb keyboard/mouse should be used. 
On Intel Compute Stick, which also has UEFI BIOS, BitKey starts  only with a very low 600x800 resolution.

The benefit of using Ubuntu prepared USB disk is that it also boots with secure boot enabled (because the grub binary is signed by Canonical).
Hope it helps.

**Comment by FetchFast** [comment]
Wish someone would run dd to make an image of this and drop it onto Mega. I can't write to the sdcard because I don't have an extfs driver :-(
